.github/workflows/default.yml

```
@@ -0,0 +1,154 @@
+        with:
+          python-version: '3.x'
+      - name: Install dependencies
+        run: |
+          python3 -m pip install --upgrade pip
+          pip3 install ansible-lint flake8 yamllint
+          which ansible
+          pip3 install ansible
+          pip3 show ansible
+          ls -l $HOME/.local/bin || true
+          ls -l /opt/hostedtoolcache/Python/3.9.1/x64/bin || true
+          echo "/opt/hostedtoolcache/Python/3.9.1/x64/bin" >> $GITHUB_PATH
+          ansible --version
+          cd $GITHUB_WORKSPACE/${{ env.ROLEPATH }}
+          { echo '[defaults]'; echo 'callback_enabled = profile_tasks, timer'; echo 'roles_path = ../'; echo 'ansible_python_interpreter: /usr/bin/python3'; } >> ansible.cfg
+          { echo '[bbb]'; echo 'localhost ansible_connection=local'; } > inventory
+          ansible-galaxy install -r requirements.yml
+      - name: Environment
+        run: |
+          pwd
+          env
+          find -ls
+      - name: run test
+        run: |
+          cd $GITHUB_WORKSPACE/${{ env.ROLEPATH }} && ansible-playbook -i inventory --become -vvv test/integration/default/default.yml ${ANSIBLE_EXTRA_VARS}
+        env:
+          PY_COLORS: '1'
+          ANSIBLE_FORCE_COLOR: '1'
+      - name: idempotency run
+        run: |
+          cd $GITHUB_WORKSPACE/${{ env.ROLEPATH }} && ansible-playbook -i inventory --become -vvv test/integration/default/default.yml ${ANSIBLE_EXTRA_VARS} | tee /tmp/idempotency.log | grep -q 'changed=0.*failed=0'  && (echo 'Idempotence test: pass' && exit 0)  || (echo 'Idempotence test: fail' && cat /tmp/idempotency.log && exit 0)
+      - name: verify
+        run: |
+          cd $GITHUB_WORKSPACE/${{ env.ROLEPATH }} && ansible-playbook -i inventory --become -v molecule/default/verify.yml ${ANSIBLE_EXTRA_VARS}
+        env:
+          PY_COLORS: '1'
+          ANSIBLE_FORCE_COLOR: '1'
+      - name: On failure
+        run: |
+          systemctl -l --no-pager status
+          systemctl -l --no-pager --failed
+          ls -l /usr/bin/ | egrep '(python|pip|ansible)'
+          pip freeze
+          pip3 freeze
+          ip addr
+          cat /etc/resolv.conf
+          host www.google.com || true
+          ping -c 1 www.google.com || true
+          ping -c 1 8.8.8.8 || true
+          host ubuntu.bigbluebutton.org || true
+          dig ubuntu.bigbluebutton.org || true
+          dig @1.1.1.1 ubuntu.bigbluebutton.org || true
+          curl -v https://ubuntu.bigbluebutton.org/repo/bigbluebutton.asc || true
+        if: ${{ failure() }}
+        continue-on-error: true
+      - name: After script - ansible setup
+        run: |
+          ansible -i inventory --connection=local -m setup localhost
+        if: ${{ always() }}
+        continue-on-error: true
+      - name: After script - firewall
+        run: |
+          sudo iptables -L -vn
+          sudo ip6tables -L -vn
+          sudo ufw status verbose
+        if: ${{ always() }}
+        continue-on-error: true
```

Personally I like to keep one liners to the top and have the long thingies at the bottom. I think its easier to read that way. (If changing that it should be changed for the whole file, the diff is only an example of what I meant)

```suggestion
      - name: After script - firewall
        if: ${{ always() }}
        continue-on-error: true
        run: |
          sudo iptables -L -vn
          sudo ip6tables -L -vn
          sudo ufw status verbose
```

---

Hmmm I understand the mindset behind that.
I will have a look.